src/databricks/labs/ucx/hive_metastore/catalog_schema.py

```diff
@@ -45,15 +45,19 @@ def _update_principal_acl(self):
             logger.debug(f"Migrating acls on {grant.this_type_and_key()} using SQL query: {acl_migrate_sql}")
             self._backend.execute(acl_migrate_sql)
 
-    def _get_catalog_schema_grants(self):
+    def _get_catalog_schema_grants(self) -> list[Grant]:
         catalog_grants: set[Grant] = set()
         new_grants = []
         src_trg_schema_mapping = self._get_database_source_target_mapping()
         grants = self._principal_grants.get_interactive_cluster_grants()
         # filter on grants to only get database level grants
-        database_grants = [grant for grant in grants if grant.table is None and grant.view is None]
+        database_grants = [
```

Comment:
rewrite as for loop, as this has just became less readable in a list comprehension